tests/test_project.py

```diff
@@ -403,3 +406,117 @@ def test_project_load_install(snippetcompiler_clean, install: bool) -> None:
         # make sure project load works after installing modules
         project.install_modules()
         project.load()
+
+
+@pytest.fixture
+def environment_icons() -> Dict[str, str]:
+    icon_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "data", "icon")
+    icon_names = ["logo.jpeg", "logo.png", "logo.svg", "logo.webp"]
+    icon_dict = {}
+    for name in icon_names:
+        icon_dict[name.split(".")[1]] = base64.b64encode(Path(os.path.join(icon_dir, name)).read_bytes()).decode("utf-8")
+    return icon_dict
+
+
+@pytest.mark.asyncio
+async def test_environment_icon_description(client_v2, environment_icons: Dict[str, str]):
+    """Test creating an environment with an icon and description"""
+
+    result = await client_v2.project_create("dev-project")
+    assert result.code == 200
+    project_id_a = result.result["data"]["id"]
+    desc = "This is an environment"
+    result = await client_v2.environment_create(project_id=project_id_a, name="env", description=desc)
+    assert result.code == 200
+    assert result.result["data"]["description"] == desc
+
+    # Test description length
+    result = await client_v2.environment_create(project_id=project_id_a, name="env2", description="a" * 256)
+    assert result.code == 400
+    result = await client_v2.environment_create(project_id=project_id_a, name="env2", description="a" * 255)
+    assert result.code == 200
+
+    for image_type, image in environment_icons.items():
+        mime_type = image_type if image_type != "svg" else "svg+xml"
+        result = await client_v2.environment_create(
+            project_id=project_id_a, name=f"env_{image_type}", description=desc, icon=f"image/{mime_type};base64,{image}"
+        )
+        assert result.code == 200
+
+    raw_icon = environment_icons["svg"]
+
+    # Test invalid icon data strings
+    result = await client_v2.environment_create(
+        project_id=project_id_a, name="envx", description=desc, icon=f"image/svg+xm;base64,{raw_icon}"
+    )
+    assert result.code == 400
+
+    result = await client_v2.environment_create(
+        project_id=project_id_a, name="envx", description=desc, icon=f"image/svg+xml;base64,{raw_icon[0:10]}"
+    )
```

Comment:
This one needs a comment to explain that `raw_icon[0:10]` results in an invalid base64 encoded string.